modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java

```diff
@@ -2224,8 +2225,11 @@ private void dumpInfo(StringBuilder sb, UUID dstNodeId) {
         else
             connPlc = new FirstConnectionPolicy();
 
+        boolean locAddrIp;
+
         try {
             locHost = U.resolveLocalHost(locAddr);
+            locAddrIp = !F.isEmpty(locAddr) && locHost.toString().startsWith("/");
```

Comment:
Can you replace this condition to a more readable one? For example, IPAddressUtil already has methods isIPv4LiteralAddress and isIPv6LiteralAddress. You also can build your own one. 
Secondly, your check is pretty dangerous. If somebody invokes locHost.getHostName() before your code, your checking is fail because getHostName has side effect.